src/test/scala/com/cognite/sdk/scala/common/RetryWhile.scala

```diff
@@ -33,4 +33,23 @@ trait RetryWhile {
       })
     result.getOrElse(action)
   }
+
+  def retryWhileEmpty[A](
+    action: => Seq[A],
+    retriesRemaining: Int = Constants.DefaultMaxRetries,
+    initialDelay: FiniteDuration = Constants.DefaultInitialRetryDelay
+  ): Seq[A] = {
+    val exponentialDelay = (Constants.DefaultMaxBackoffDelay / 2).min(initialDelay * 2)
+    val randomDelayScale = (Constants.DefaultMaxBackoffDelay / 2).min(Constants.DefaultInitialRetryDelay * 2).toMillis
+    val nextDelay = Random.nextInt(randomDelayScale.toInt).millis + exponentialDelay
+    val result : Seq[A] = action
+    if (!result.isEmpty) {
+      result
+    } else if (retriesRemaining > 0) {
+      Thread.sleep(initialDelay.toMillis)
+      retryWhileEmpty[A](action, retriesRemaining - 1, nextDelay)
+    } else {
+      throw new AssertionError(s"Retry count exceeded, result is still empty.")
+    }
+  }
```

Comment:
Let's avoid duplicating this entire function. Can we introduce a simpler wrapper over `retryWithExpectedResult` instead?

I'm thinking something like

```
def retryWhile[A](
    action: => A,
    assertion: A => Assertion,
    retriesRemaining: Int = Constants.DefaultMaxRetries,
    initialDelay: FiniteDuration = Constants.DefaultInitialRetryDelay
): A = retryWithExpectedResult(action, None, Seq(assertion), retriesRemaining, initialDelay)
```

Reply:
That could be, but I still like that with the signature `retryWhileEmpty(action: => A)` you can just write `retryWhileEmpty { ... }`, without any other parameters that are after the entire retried expression.